mesheryctl/pkg/utils/platform_test.go

```diff
@@ -0,0 +1,596 @@
+			}
+
+			// Actual file contents
+			actualContent, err := ioutil.ReadFile(testConfigPath)
+			if err != nil {
+				t.Error("Error reading actual file contents: ", err)
+			}
+
+			actualFileContent := string(actualContent)
+
+			// Expected file contents
+			testdataDir := currDir + "/testdata/platform"
+
+			golden := NewGoldenFile(t, tt.golden, testdataDir)
+			if *update {
+				golden.Write(actualFileContent)
+			}
+			expectedFileContent := golden.Load()
+
+			if expectedFileContent != actualFileContent {
+				t.Errorf("Expected file content \n[%v]\n and actual file content \n[%v]\n don't match", expectedFileContent, actualFileContent)
+			}
+
+			// Repopulating Expected yaml
+			if err := Populate(currDir+"/fixtures/platform/original/TestChangeEndpointConfig.yaml", testConfigPath); err != nil {
+				t.Fatal(err, "Could not complete test. Unable to repopulate fixture")
+			}
+		})
+	}
+}
+
+func TestGetManifestURL(t *testing.T) {
+	type args struct {
+		manifest        Manifest
+		rawManifestsURL string
+	}
+
+	rawMfestsURL := "https://testurl.com/k8s/"
+
+	tests := []struct {
+		name    string
+		args    args
+		fixture string
+		golden  string
+	}{
+		{
+			name: "Get Manifest URL (blob)",
+			args: args{
+				rawManifestsURL: rawMfestsURL,
+			},
+			fixture: "getmanifesturl.blob.golden",
+			golden:  "getmanifesturl.expect.blob.golden",
+		},
+		{
+			name: "Get Manifest URL (non-blob)",
+			args: args{
+				rawManifestsURL: rawMfestsURL,
+			},
+			fixture: "getmanifesturl.other.golden",
+			golden:  "getmanifesturl.expect.other.golden",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Read fixture
+			file, _ := ioutil.ReadFile("fixtures/platform/" + tt.fixture)
+			manifest := Manifest{}
+			_ = json.Unmarshal([]byte(file), &manifest)
+
+			tt.args.manifest = manifest
+
+			got := GetManifestURL(tt.args.manifest, tt.args.rawManifestsURL)
+
+			// Read golden file
+			golden := NewGoldenFile(t, tt.golden, "testdata/platform/")
+			if *update {
+				golden.Write(got)
+			}
+
+			want := golden.Load()
+
+			if got != want {
+				t.Fatalf("GetManifestURL() = %v, want %v", got, want)
+			}
+		})
+	}
+}
+
+func TestIsPodRequired(t *testing.T) {
+	type args struct {
+		requiredPods []string
+		pod          string
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "Pod '" + "test'" + " is required",
+			args: args{
+				requiredPods: []string{"test", "test1", "test2"},
+				pod:          "test",
+			},
+			want: true,
+		},
+		{
+			name: "Pod '" + "test'" + " is not required",
+			args: args{
+				requiredPods: []string{"test1", "test2"},
+				pod:          "test",
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := IsPodRequired(tt.args.requiredPods, tt.args.pod); got != tt.want {
+				t.Fatalf("IsPodRequired() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
```

Comment:
Let's not write tests for this function and shift `IsPodRequired` to `logs.go`

Reply:
as the function isn't being used anywhere other than this file. and it's best to clean out the code from here, and this function will be covered from logs.go